auth/client-provider/src/test/java/org/wildfly/security/auth/client/provider/DefaultSSLContextNonexistentConfigFileTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.client.provider;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.wildfly.security.auth.client.AuthenticationContext;
+
+import javax.net.ssl.SSLContext;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+
+/**
+ * Test that default SSLContext provider will throw an exception when configured file path does not exist
+ */
+public class DefaultSSLContextNonexistentConfigFileTest {
+
+    private static final String CONFIG_FILE = "./src/test/resources/org/wildfly/security/auth/client/provider/wildfly-config-invalid-path.xml";
+
+    @Test(expected = IllegalArgumentException.class)
+    public void defaultSSLContextNonexistentConfigFileTest() {
+        Security.insertProviderAt(new ElytronClientDefaultSSLContextProvider(CONFIG_FILE), 1);
+        Assert.assertNotNull(Security.getProvider("ElytronClientDefaultSSLContextProvider"));
+        AuthenticationContext authenticationContext = AuthenticationContext.captureCurrent();
+        authenticationContext.run(() -> {
+            SSLContext defaultSSLContext = null;
+            try {
+                defaultSSLContext = SSLContext.getDefault();
+            } catch (NoSuchAlgorithmException e) {
+                Assert.fail("Obtaining default SSL context from provider with invalid path threw incorrect exception");
+            }
+            Assert.assertNotNull(defaultSSLContext);
+            Assert.assertNotEquals(ElytronClientDefaultSSLContextProvider.class.getSimpleName(), defaultSSLContext.getProvider().getName()); // diff provider was used since elytron provider is looping
+            Assert.assertNotNull(defaultSSLContext.getSocketFactory());
```

Comment:
So then these assertions won't actually be hit in this test.

Reply:
Correct, I removed those assertions